src/pkg/cli/compose/context.go

```diff
@@ -340,6 +340,16 @@ func WalkContextFolder(root, dockerfile string, fn func(path string, de os.DirEn
 		return err
 	}
 
+	if dockerignore == "" && writeIgnore {
+		// Generate a default .dockerignore file if none exists (to be included in the context)
+		term.Warn("No .dockerignore file found; creating default .dockerignore")
+		var err error
+		dockerignore, err = writeDefaultIgnoreFile(root, dotdockerignore)
+		if err != nil {
+			return fmt.Errorf("failed to write default .dockerignore file: %w", err)
+		}
+	}
+
```

Comment:
I would promote this to `createArchive` instead of adding a conditional parameter to this function.

```suggestion
```

Reply:
I agree this nicer, but then I have to duplicate the `getDockerIgnorePatterns` code.